code/z_adventurezones/safehouse.dm

```diff
@@ -0,0 +1,686 @@
+		bioHolder.mobAppearance.underwear = "none"
+		bioHolder.mobAppearance.u_color = "#FFFFFF"
+		bioHolder.Uid = bioHolder.CreateUid()
+		bioHolder.uid_hash = md5(bioHolder.Uid)
+		. = ..()
+
+obj/item/reagent_containers/iv_drip/dead_exec
+	desc = "A bag filled with someone's blood. It's labelled 'Jean Rockefeller'."
+	icon_state = "IV-blood"
+	mode = 1
+	initial_reagents = "blood"
+
+/obj/machinery/handscanner/bio_handscanner
+	name = "Hand Scanner"
+	desc = "A biometric hand scanner."
+	icon = 'icons/obj/decoration.dmi'
+	icon_state = "handscanner"
+	id = "clone room"
+	var/allowed_bioHolders = null
+	var/cooldown = FALSE
+
+/obj/machinery/handscanner/bio_handscanner/attackby(obj/item/W, mob/user as mob) //We don't want the parents proc here.
+	if(istype(W, /obj/item/device/detective_scanner))
+		return
+	if(istype(W, /obj/item/parts/human_parts/arm/))
+		boutput(user, "<span class='alert'>ERROR: no pulse detected.</span>")
+	if(istype(W, /obj/item/card/emag))
+		for(var/obj/machinery/door/poddoor/M in by_type[/obj/machinery/door])
+			boutput(user, "You short out the hand scanner's circuits. So much for cutting edge.")
+			if(M.id == src.id)
+				if(M.density)
+					M.open()
+				else
+					M.close()
+	return src.Attackhand(user)
+
+/obj/machinery/handscanner/bio_handscanner/attack_hand(mob/user as mob) //Nor here.
+	src.add_fingerprint(user)
+	if(cooldown == TRUE)
+		return
+	playsound(src.loc, "sound/effects/handscan.ogg", 50, 1)
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.bioHolder.Uid == allowed_bioHolders) //Are you the authorised bioHolder (for all intents and purposes)?
+			user.visible_message("<span class='notice'>The [src] accepts the biometrics of the hand and beeps.</span>")
+			for(var/obj/machinery/door/poddoor/M in by_type[/obj/machinery/door])
+				if(M.id == src.id)
+					if(M.density)
+						M.open()
+					else
+						M.close()
+		else
+			boutput(user, "<span class='alert'>Invalid biometric profile. Access denied.</span>")
+		cooldown = TRUE
+		sleep(1 SECOND) //To reduce chat spam in case of multi-click.
+		cooldown = FALSE
+
+// CLONESCAN HEALTH IMPLANT
+
+/obj/item/implant/health/exp_health_implant
+	name = "experimental health implant"
+	desc = ""
+	icon_state = "implantpaper-b"
+	impcolor = "b"
+	var/clonescanned = 0
+
+	activate()
+		..()
+		clonescanned = 0
+
+	do_process(var/mult = 1) //On do_process so we don't have to worry if the user is dead / in ghostVR.
+		if (!ishuman(src.owner))
+			return
+		var/mob/living/carbon/human/H = src.owner
+		if (clonescanned < 11)
+			switch(clonescanned++)
+				if(1)
+					boutput(H,"Life signs detected. Initiating scanning procedure.")
+				if(4)
+					boutput(H,"Scan complete. Contacting remote server for data upload.")
+				if(7)
+					boutput(H,"ERROR: Remote server unavailable. Commencing emergency transmisssion protocols. Broadcasting data on all known frequencies.")
+				if(10)
+					for(var/obj/machinery/computer/cloning/C in world)
```

Comment:
Comment from pali on discord
*the for( ... in world) loops are very slow (because they iterate through everything in the world), same with loops of the form for(var/some/type/variable)
a good way to avoid these is to put START_TRACKING in your type's New and STOP_TRACKING in its disposing() and then use
for_by_tcl(variable, /some/type) which will be much faster*

Reply:
Replaced main loops with for_by_tcl. Loops that use for(bytype) on poddoors remain, as they're generic poddoors and/or they're using their parent's proc's.